src/helper/children-loader.js

```diff
@@ -0,0 +1,112 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+/**
+ * Configures the children component definition, allows this component to determine
+ * which components to load in a dynameic way using the parameter passed along
+ * from the load function of the root component
+ */
+define([
+            'jquery',
+            'ju-shared/class'
+        ],
+        function(
+            $,
+            Class
+        ) {
+    'use strict';
+
+    // Enable a base component to load children dynamically (attach methods) based on
+    // custom logic like permissions or flags. Review following util methods:
+
+    // ---- sample child definition ----
+    // component_name : {
+    //     component : Catalog.COMPONENT,
+    //     insertionPoint : '.selector',
+    //     [...flags],
+    //     extendedOpts : {...}
+    // }
+
+    // @interface: loadChildrenDefinition
+
+    // @warn ONLY initChildrenDef should have access to childrenKey and _addChildrenDef
+
+    var ChildrenLoaderHelper = Class.extend({
+        init : function(childrenKeys, childrenDefinitions) {
+            // Define childrens
+            this.childrenDef = this.childrenDef || {};
+            this.childrenDefinitions = childrenDefinitions || {};
+
+            // Loaded children keys
+            // Usually a set of default keys is included at initChildrenDef
+            this.childrenKeys = childrenKeys || [];
+        },
+        /**
+         * Defines if a children is accepted from the definition
+         * Custom implementation must be provided by project
+         * @abstract
+         */
+        loadChildDefinition : function(/* key, definition */) {},
```

Comment:
what if instead of an `abstract` method we inject a callback function? this way we can discourage the usage of inheritance for children loading (you still may add a separate file with the callback setup if you want)